acceptance-tests/remote.go

```diff
@@ -0,0 +1,236 @@
+package acceptance_tests
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"time"
+
+	"github.com/bramvdbogaerde/go-scp"
+	"golang.org/x/crypto/ssh"
+)
+
+// runs command on remote machine
+func runOnRemote(user string, addr string, privateKey string, cmd string) (string, string, error) {
+	client, err := buildSSHClient(user, addr, privateKey)
+	if err != nil {
+		return "", "", err
+	}
+
+	session, err := client.NewSession()
+	if err != nil {
+		return "", "", err
+	}
+	defer session.Close()
+
+	var stdOutBuffer bytes.Buffer
+	var stdErrBuffer bytes.Buffer
+	session.Stdout = &stdOutBuffer
+	session.Stderr = &stdErrBuffer
+	err = session.Run(cmd)
+	return stdOutBuffer.String(), stdErrBuffer.String(), err
+}
+
+func copyFileToRemote(user string, addr string, privateKey string, remotePath string, fileReader io.Reader, permissions string) error {
+	clientConfig, err := buildSSHClientConfig(user, addr, privateKey)
+	if err != nil {
+		return err
+	}
+
+	scpClient := scp.NewClient(fmt.Sprintf("%s:22", addr), clientConfig)
+	if err := scpClient.Connect(); err != nil {
+		return err
+	}
+
+	return scpClient.CopyFile(context.Background(), fileReader, remotePath, permissions)
+}
+
+func copyFileFromRemote(user string, addr string, privateKey string, remotePath string, localFile *os.File, permissions os.FileMode) error {
```

Comment:
This is wrapped with `downloadFile` in the bosh_helper.go file.

Is there a scenario where we would issue a `copyFileFromRemote()` without having a `pcapInfo`?

Reply:
yes, there is. `downloadFile` currently implies `pcap-api` as the target. If you want to copy stuff from somewhere else you would use `copyFileFromRemote`. Granted, `downloadFile` is a convenience function.